controllers/result.go

```
@@ -29,15 +30,28 @@ func (c *ResultController) List() {
 
 func (c *ResultController) Create() {
 	c.EnsureAuthenticatedUser()
-
-	keywords := []string{
-		"ergonomic chair",
-		"cloud storage service",
-		"cloud computing service",
-		"crypto currency",
-		"เตา balmuda",
+	flash := web.NewFlash()
+
+	file, fileHeader, err := c.GetFile("file")
+	if err != nil {
+		flash.Error("Failed to upload file, please make sure the file is not corrupted")
+	} else {
+		uploadForm := forms.UploadForm{
+			File: file,
+			FileHeader: fileHeader,
+			User: c.CurrentUser,
+		}
+		keywords, errs := uploadForm.Save()
+		if len(errs) > 0 {
+			for _, err := range errs {
+				flash.Error(err.Error())
```

It will just show the last flash error, right? If so maybe we can just return a single error from `uploadForm.Save()` 🤔

---

Exactly! updated in [47d92bf](https://github.com/carryall/go-google-scraper-challenge/pull/56/commits/47d92bf0b0bfc660de2a408aba7a639c5e92ca7f)